junit-jupiter-params/src/test/java/org/junit/jupiter/params/ParameterizedTestExtensionTests.java

```diff
@@ -228,4 +244,21 @@ void method(String parameter) {
 		}
 	}
 
+	static class TestCaseWithArgumentSourceAnnotatedMethodWithNonStaticArgumentProvider {
+
+		@ParameterizedTest
+		@ArgumentsSource(NonStaticArgumentProvider.class)
+		void method(Collection<String> parameters) {
+		}
+
+	}
+
+	class NonStaticArgumentProvider implements ArgumentsProvider {
```

Comment:
Please introduce an additional test for a static nested class that does not contain a no-argument constructor.